packages/fxa-settings/src/models/reliers/oauth-relier.test.ts

```diff
@@ -25,5 +25,135 @@ describe('models/reliers/oauth-relier', function () {
     expect(model).toBeDefined();
   });
 
+  describe('scope', () => {
+    const SCOPE = 'profile:email profile:uid';
+    const SCOPE_PROFILE = 'profile';
+    const SCOPE_PROFILE_UNRECOGNIZED = 'profile:unrecognized';
+    const SCOPE_WITH_PLUS = 'profile:email+profile:uid';
+    const SCOPE_WITH_EXTRAS =
+      'profile:email profile:uid profile:non_whitelisted';
+    const SCOPE_WITH_OPENID = 'profile:email profile:uid openid';
+
+    function getRelierWithScope(scope: string) {
+      const relier = new OAuthRelier(
+        new GenericData({
+          scope,
+        }),
+        new GenericData({}),
+        {
+          scopedKeysEnabled: true,
+          scopedKeysValidation: {},
+          isPromptNoneEnabled: true,
+          isPromptNoneEnabledClientIds: [],
+        }
+      );
+
+      return relier;
+    }
+
+    describe('is invalid', () => {
+      function getRelier(scope: string) {
+        return getRelierWithScope(scope);
+      }
+
+      it('empty scope', () => {
+        const relier = getRelier('');
+        expect(() => relier.getPermissions()).toThrow();
+      });
+
+      it('whitespace scope', () => {
+        const relier = getRelier(' ');
+        expect(() => relier.getPermissions()).toThrow();
+      });
+    });
+
+    describe('is valid', () => {
+      function getRelier(scope: string) {
+        return getRelierWithScope(scope);
+      }
```

Comment:
~Q: how does this work? I see we have this function defined already in L55-57, but I thought functions were hoisted up into a higher scope, so not sure if this is just overriding the previous function or if they're just hoisted slightly higher.~ Although I suppose I could verify that locally with a simpler test case and some console logs.

Once again, I was wrong! Nested functions seem to be hoisted to the top of THEIR scope, not into the top scope, so a function defined within `test("one", () => {})` won't be shared w/ a different `test()`:

```js
describe("stuff", () => {
  test("one", () => {
    expect(getScope()).toBe("oNe");

    function getScope(scope = "oNe") {
      console.log(`getScope() "${scope}"`);
      return scope;
    }
  });

  test("two", () => {
    // "ReferenceError: getScope is not defined"
    expect(getScope()).toBe("tWo");

    // function getScope(scope = "tWo") {
    //   console.log(`getScope() "${scope}"`);
    //   return scope;
    // }
  });
});
```